### DISCUSSION TITLE
Need clarification on creating an APP

### DISCUSSION THREAD

**Discussion by MadhuGowthaman** [mastodon/mastodon#24226] [Q&A] [answered]
Hello,

Do we need to create an application for each instance (server)? or does creating a single app sufficient to communicate with all instances?

Please give clarification.

Thanks,
Madhubala.

**Comment by MadhuGowthaman** [comment]
Team, could you kindly respond? Or is there somewhere else I can ask or get support from?

**Comment by DejavuMoe** [comment, marked as answer]
It is not clear whether you want to create a web application (OAuth2) or a client application, since Mastodon opens up a unified API, I thought it would be better to create an application that supports communication with all instances. (user enters instance URL, login credentials are available).

>The developer team seems to be dealing with quite a lot of discussions and topics on a daily basis, so we may not get a quick response. I hope my answer will be helpful to you, have a nice day!

**Reply by MadhuGowthaman** [reply]
Hi DejavuMoe,

Thanks for your reply. 

I am asking about [this](https://docs.joinmastodon.org/methods/apps/#create) app creation. Do we need to create a different app for each instance or will a single app be able to communicate with all instances, is my question.

Example: Can I communicate with Mastodon.world using an app I create for Mastodon.social?

Thanks,
Madhubala

**Reply by Gargron** [reply]
No, your credentials are valid for one instance only.

**Reply by MadhuGowthaman** [reply]
@Gargron Thank you so much for your response.
